### PR TITLE
[#2627] improve (trino-connector): Support loading catalogs before the Trino started

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

 Make  loading catalogs before the Trino started

### Why are the changes needed?

Fix: #2627 

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

UT

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java [hunk 1]
```diff
@@ -101,31 +102,40 @@ public void start(GravitinoAdminClient client) {
...
+      } catch (NoSuchMetalakeException noSuchMetalakeException) {
+        LOG.warn("Metalake {} does not exist.", usedMetalake);
+        continue;
+      } catch (Exception e) {
+        LOG.error("Load Metalake {} failed.", usedMetalake, e);
+        continue;
+      }
+
+      LOG.info("Load metalake: {}", usedMetalake);
+      loadCatalogs(metalake);
```
[the remark this answers is not in the document]
Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java]: Fixed

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Comment by yuqi1129** [comment]
@diqiu50 
Should this PR  be merged into branch-0.5?

**Comment by diqiu50** [comment]
> @diqiu50 Should this PR be merged into branch-0.5?

No need

**Pull request merged**